### PR TITLE
Add import statement ordering

### PR DESCRIPTION
**Pull request by mbj**
### Acceptance

* [x] Sucessful `stack test`

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb, @snusnu This is ready for review.

File: src/SourceConstraints.hs [hunk 1]
```diff
@@ -191,3 +203,34 @@ render dynFlags outputable =
...
+
+    candidates :: [(String, String, Located a)]
+    candidates =
+      let rendered = render dynFlags <$> nodes
+      in
+        zip3 rendered (sort rendered) nodes
+
+zip3 :: [a] -> [b] -> [c] -> [(a, b, c)]
+zip3 (a:as) (b:bs) (c:cs) = (a, b, c) : zip3 as bs cs
+zip3 _      _      _      = []
```
Comment (dkubb) [inline src/SourceConstraints.hs]: What about using `zip3` from `Data.List`? It looks like you import that module in this code already.

Comment (mbj) [reply src/SourceConstraints.hs]: Hah, I did not think haskell has a build in one.

**Comment by mbj** [comment]
@dkubb, @snusnu This is ready for the next pass.

File: src/SourceConstraints.hs [hunk 2]
```diff
@@ -2,26 +2,26 @@
...
+import Bag (emptyBag, unitBag, unionManyBags)
 import Control.Applicative (Alternative(empty), Applicative(pure))
 import Control.Monad (when)
 import Control.Monad.IO.Class (MonadIO(liftIO))
 import Data.Bool (Bool(False), not)
 import Data.Char (isUpper)
 import Data.Data (Data, Typeable, cast, gmapQ)
 import Data.Eq (Eq((/=)))
-import Data.Foldable (elem)
+import Data.Foldable (elem, find)
```
Comment (dkubb) [inline src/SourceConstraints.hs]: Not a blocker, but `elem` is a method on `Foldable`.

**Comment by snusnu** [comment]
@mbj I'd reword

* `Fix import statement ordering to be canonical` => `Fix SourceConstraints import order`
* `Refactor to prefer maybe function over pattern match` => `Refactor SourceConstraints.runSourceConstraints to be simpler`

(I am not so sure about the latter one)

**Comment by mbj** [comment]
> @mbj I'd reword

I moved the discussion to the channel.

**Comment by mbj** [comment]
@dkubb, @snusnu This is ready for the next pass.
